lifecycle-operator/controllers/common/eventsender.go

```diff
@@ -8,19 +8,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type EventSender struct {
+//go:generate moq -pkg fake -skip-ensure -out ./fake/event_mock.go . IEvent:MockEvent
+type IEvent interface {
+	SendEvent(phase apicommon.KeptnPhaseType, eventType string, reconcileObject client.Object, status string, message string, version string)
```

Comment:
nit: As the interface is already called `IEvent`, we could rename the function to `Send`, or maybe `Emit`, to make it less verbose

Reply:
👍 good point!